### ISSUE TITLE
Bug: Date picker field is broken and crashes (Tinasaurus)

### ISSUE THREAD

**Issue by hrueger** [tinacms/tinasaurus#10]
### Describe the issue that you're seeing. Any Loom videos or screenshots usually help a lot!

I'm using the [base tinasaurus](https://github.com/tinacms/tinasaurus) project. When editing a post of the blog and changing the `Date` field, behaviour is as follows:
- When selecting a date, it shows `required` and `Save` is disabled:
![grafik](https://github.com/tinacms/tinacms/assets/18656830/3f934372-ee69-4242-bfd5-0d137aaa947b)
- When deleting the field content (Ctrl+A and Delete) and typing (e.g. `s` for `september`), the whole page crashes and we get a `date value is not finite in DateTimeFormat.format()`:
![grafik](https://github.com/tinacms/tinacms/assets/18656830/fad40fda-70a5-42a5-bef5-4eaa280539a5)

### Reproduction

https://github.com/tinacms/tinasaurus

### Steps to reproduce

Clone the tinasaurus project, start it and edit the date field of a blog post (see above on what errors to expect).

### System Info

```shell
System:
    OS: Windows 10 10.0.22621
    CPU: (16) x64 12th Gen Intel(R) Core(TM) i7-1260P
    Memory: 4.24 GB / 15.62 GB
  Binaries:
    Node: 18.16.1 - C:\Program Files\nodejs\node.EXE
    Yarn: 1.22.19 - C:\_Source\docs\node_modules\.bin\yarn.CMD
    npm: 9.5.1 - C:\Program Files\nodejs\npm.CMD
  Browsers:
    Edge: Spartan (44.22621.2215.0), Chromium (116.0.1938.62)
    Internet Explorer: 11.0.22621.1
```

### Validations

- [X] Follow our [Code of Conduct](https://github.com/tinacms/tinacms/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [docs](https://tina.io/docs).
- [X] Check that there isn't [already an issue](https://github.com/tinacms/tinacms/issues) that reports the same bug.

**Comment by jamespohalloran** [comment]
I just opened a [PR](https://github.com/tinacms/tinasaurus/pull/9) that upgrades Tina, but also fixes an issue with the date field in posts. 
To fix in your local repo, you can probably just update this line: https://github.com/tinacms/tinasaurus/pull/9/files#diff-f49256b71dfc7711f6f7e7e91d2624e8a13679b215db8f75ee227700a9451822R111

**Comment by hrueger** [comment]
Hi @jamespohalloran,
thanks for the response. I did the same changes in my repo, selecting a field in the datepicker works now.
However, problem 2 (crash when typing) is still an issue...

**Comment by jamespohalloran** [comment]
@hrueger  Have you tried upgrading the Tina packages like I've done in the PR as well? I'm unable to reproduce the crash, but could be a Windows-specific thing

**Comment by hrueger** [comment]
Yes, I did. Here's the project: https://github.com/Dein-Ticket-Shop/docs/
When I select everything in the field, clear it and then type a letter, it crashes.
I don't think it is windows-related since it is a clientside js error. Do you need a video?

**Comment by jamespohalloran** [comment]
@hrueger A video would be helpful if you wouldn't mind!

**Comment by hrueger** [comment]
Hi @jamespohalloran,
sure, no problem. Here you go: https://github.com/tinacms/tinasaurus/assets/18656830/2473dc74-27f3-4250-b557-82f833ef4634

just before it crashed, I was typing an "a". You can't see that in the video.

Does that help?
